### ISSUE TITLE
Inconsistent deserialize of DateTime on Xamarin Android

### ISSUE THREAD

**Issue by MattiaDurli** [JamesNK/Newtonsoft.Json#2548]
On a SQLServer database table I have two birth dates:

1975-10-11 00:00:00.000

1983-07-19 00:00:00.000

A service (not mine) exposes the birthdates as JSON, with DateTime in the following format (I can't change it):

"\/Date(182210400000-0000)\/"

"\/Date(427413600000-0000)\/"

The basic code for test:

    response = await client.PostAsync(uri, contentReq);
    if (response.IsSuccessStatusCode)
                    {
                        var contentResp = await response.Content.ReadAsStringAsync();
                        Console.WriteLine("JSON:" + contentResp);
    
                        var user = JsonConvert.DeserializeObject<User>(contentResp);
    
    
                        TimeSpan ts1 = new DateTimeOffset(user.BirthDate.Value).Offset;
                        Console.WriteLine($"Offset:{ts1.ToString()}");
                        Console.WriteLine($"Data:{user.BirthDate.Value.ToString()}");
    }

Output on .NET5 Console app:

Offset:02:00:00

Date:11/10/1975 00:00:00

Offset:02:00:00

Date:19/07/1983 00:00:00

Output on a Xamarin Android App:

**Offset:01:00:00**

**Date:10/10/1975 23:00:00**

Offset:02:00:00

Date:19/07/1983 00:00:00

Please help me understand the Xamarin error in deserialization. Why on just one of the values?
I tried other dates and times, and result is unpredictable.

UPDATE1:

tested with and without

     JsonSerializerSettings microsoftDateFormatSettings = new JsonSerializerSettings
                {
                    DateFormatHandling = DateFormatHandling.MicrosoftDateFormat
                };

and adding 

    var user = JsonConvert.DeserializeObject<User>(contentResp,microsoftDateFormatSettings);

### Source/destination types

```csharp
 public class User
    {
        public DateTime? BirthDate { get; set; }
}
```
### Source/destination JSON

{"BirthDate":"\/Date(182210400000-0000)\/"}

### Expected behavior

Consistent DateTime serialization on all platforms

### Actual behavior

On .NET5 Console, correct deserialization, on Xamarin Android timezones are incorrect, but inconsistently

### Steps to reproduce

```csharp
response = await client.PostAsync(uri, contentReq);
    if (response.IsSuccessStatusCode)
                    {
                        var contentResp = await response.Content.ReadAsStringAsync();
                        Console.WriteLine("JSON:" + contentResp);
    
                        var user = JsonConvert.DeserializeObject<User>(contentResp);
    
    
                        TimeSpan ts1 = new DateTimeOffset(user.BirthDate.Value).Offset;
                        Console.WriteLine($"Offset:{ts1.ToString()}");
                        Console.WriteLine($"Data:{user.BirthDate.Value.ToString()}");
    }
```

**Comment by elgonzo** [comment]
That is not an issue with Json.NET, as far as i can tell.

Pay attention to the day/month of the dates. What you observe fits perfectly when using a region/time-zone that uses daylight saving.

**Comment by MattiaDurli** [comment]
great. Now I can understand the timezone difference on the same platform.
But then how can I get rid of the timezone on Xamarin Android and have the same calculation I have on my desktop console app?
My Windows10 and Android devices are both in Italian and GMT+2, why this winter time zone calculation on Android?

**Comment by elgonzo** [comment]
I guess the Json.NET issue tracker is not the right place to discuss time-zone related issues.

However, with regard to your observation: I don't know, but it almost looks to me like .NET 5's implementation does not "know" that the summertime window for Italy was different in 1975 (from  beginning of June to  end of September) than it is toady (from end of March to end of October). But i am just speculating here, don't take my word for it...

Ideally i would suggest you (or whoever is in charge of the server-side code) change the serialization on the server-side so that the serialized timestamps are including a time-zone offset, so your deserialization doesn't rely on the client _just so_ having the correct timezone configured.

**Comment by elgonzo** [comment]
Well, i didn't notice before that you seem to be only interested in birth dates, so only day, month, year, but not the actual time in hours/minutes/seconds. If that's the case, timezone shouldn't matter at all, and there should be a simple solution for you:

Treat the timestamps in the JSON as UTC times. You can instruct Json.NET to deserialize the timestamps as UTC by setting the `DateTimeZoneHandling` parameter in the `JsonSerializerSettings` to `DateTimeZoneHandling.UTC`.

**Comment by elgonzo** [comment]
I looked into it a bit closer. The issue is indeed not caused by Json.NET. The observed problem can be demonstrated entirely without Json.NET.

See this code example on dotnetfiddle: https://dotnetfiddle.net/LCnkot The dotnetfiddle service is running on some Linux or such. Note the difference in timezone offset.

Now, take the very same code and copy&paste it into a console application project for your Windows box. Change the _timeZone field initialization to "Central European Standard Time" (because that is the CET identifier as used on Windows). Then run it, and you notice the timezone offsets remain both at 2:00.

Note that this little demo program does NOT use Json.NET, demonstrating that this issue is unrelated to Json.NET. Which means, this issue here should be closed (and, if there is further need for help, sites like stackoverflow.com provide an excellent platform for asking about problems with C# or .NET outside of Json.NET.)

**Comment by MattiaDurli** [comment]
Thanks for your analysis, but I investigated further and the issue is not even related to that, but more to the MicrosoftDateFormat deserialization.

The two dates are converted on the server to UTC and sent in MicrosoftDateFormat, with 2 hours less, as we are in GMT+2 DST now (Italy GMT+1, +1 DST)
I converted both utc datetimes (182210400000 and 427413600000) on an online converter and are both GMT 22:00PM of the day before.
And it's ok so far.

Now the UTC in MicrosoftDateFormat is deserialized on the two clients, both in GMT +2 DST.
On Windows .NET5 (but I guess the older .NETs), since we are in GMT+2 DST, 2 hours are added to both dates and we're back to the correct DateTime.
On Android instead, the conversion takes into account if the date to convert is in a DST period, and if it is so, converts it to GMT +2 DST, adding 2 hours, and if not, it adds only 1 hour and sets it to GMT +1.

My question to you, for me to better understand this difference of behavior, is: 
who takes care of the conversion of the MicrosoftDateFormat to DateTime? a .NET/Mono function, or JSON.NET?

**Comment by elgonzo** [comment]
It's not Json.NET's fault as far as i can tell, as the issue is reproducible **without** Json.NET. (If Json.NET were to blame, the issue should _not_ be reproducible without Json.NET.) I guess the issue is related to the underlying OS (again, look at the dotnetfiddle i provided in https://github.com/JamesNK/Newtonsoft.Json/issues/2548#issuecomment-863437928. This dotnetfiddle is running as a .NET 5 app on a Linux/Unix-like box. The same code yields different offsets, when run as a .NET 5 console app on a Windows box.)

Also, the timestamps in your JSON do not carry any information about them being UTC. You know they are representing UTC time, but how should your program and the deserializer know this if you don't tell them? Remember, the timestamps in your JSON do not carry information that identifies them as being UTC.

You seem to incorrectly assume that strings should by default be converted to UTC-based DateTime instances when the string lacks a UTC specifier or timezone offset. This assumption is wrong. Json.NET just follows the default behavior of `DateTime`: using the local timezone when parsing a date/time string without timezone offset nor UTC specifier.

On the upside, since you have confirmed that the timestamps in your JSON are in UTC, you can instruct the (de)serializer to assume UTC for all timestamps without timezone offset nor UTC specifier. See my comment https://github.com/JamesNK/Newtonsoft.Json/issues/2548#issuecomment-863419828 about how you can do that.

**Comment by MattiaDurli** [comment]
Yes it's clear that's not json.net by now, and not even the MicrosoftDateFormat. 
I already managed the UTC decoding and timezone conversion from GMT, but my problem was not the timezone in itself, but the DST into the same timezone: GMT+1 and GMT+2 DST (that is different from GMT+2, albeit being the same +2 hours).

After going through your explanation and other suggestions I now fixed it my problem by detecting the offset between the current GMT and the calculated one.

What still strucks me is that starting from a pure UTC value, on a Windows .NET system, date is created adding the current GMT with any date, so in my case always adds 2H and sets it at GMT+2 DST
On a Android/Linux/MONO system, date is created by adding the GMT of the timezone at that date, so if I process a winter date, it adds GMT +1, if is a summer date it adds GMT +2 DST

Whatever, I solved my problem, thanks anyway.
